src/cbc_sdk/platform/alerts.py

```diff
@@ -1496,3 +1552,66 @@ def set_remote_is_private(self, is_private, exclude=False):
         else:
             self._exclusions["remote_is_private"] = is_private
         return self
+
+
+class GroupedAlertSearchQuery(AlertSearchQuery):
+    """Represents a query that is used to group Alert objects by a given field."""
+    def __init__(self, *args, **kwargs):
+        """Initialize the GroupAlertSearchQuery."""
+        super().__init__(*args, **kwargs)
+        self._group_by = "THREAT_ID"
+
+    def group_by(self, field):
+        """
+        Sets the 'group_by' query body parameter, determining which field to group the alerts by.
+
+        Args:
+            field (string): The field to group by
+        """
+        self._group_by = field
+        return self
+
+    def _perform_query(self, from_row=1, max_rows=-1):
+        """
+        Performs the query and returns the results of the query in an iterable fashion.
+
+        Args:
+            from_row (int): The row to start the query at (default 1).
+            max_rows (int): The maximum number of rows to be returned (default -1, meaning "all").
+
+        Returns:
+            Iterable: The iterated query.
+        """
+        url = self._build_url("/_search")
+        current = from_row
+        numrows = 0
+        still_querying = True
+        while still_querying:
+            request = self._build_request(current, max_rows)
+            request["group_by"] = {"field": self._group_by}
+            resp = self._cb.post_object(url, body=request)
+            result = resp.json()
+
+            self._total_results = result["num_found"]
+            self._group_by_total_count = result["group_by_total_count"]
+
+            # Prevent 500 Internal Server Error from retrieving behind MAX_RESULTS_LIMIT
+            if self._total_results > MAX_RESULTS_LIMIT:
+                self._total_results = MAX_RESULTS_LIMIT
+            self._count_valid = True
+
+            results = result.get("results", [])
+            for item in results:
+                grouped_alert = self._doc_class(self._cb, None, item)
+                yield grouped_alert
+                current += 1
+                numrows += 1
+
+                if max_rows > 0 and numrows == max_rows:
+                    still_querying = False
+                    break
+
+            from_row = current
```

Comment:
from_row is not used after this assignment.

Reply:
I think this is part of the generator so if its called again it knows where it left off